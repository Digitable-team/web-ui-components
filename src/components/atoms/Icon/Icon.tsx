import React from 'react'

import { Icons } from 'src/components/atoms/Icon/Icon.types'

import { getIcon } from 'src/components/atoms/Icon/Icon.utils'

export type IconProps = {
  name: Icons | string
  tooltip?: string
  onClick?: () => void
  className?: string
}

export const Icon = ({ name, onClick, className, tooltip }: IconProps) => {
  const IconComponent = getIcon(name)

  return (
    <div className={className} onClick={onClick} title={tooltip}>
      <IconComponent />
    </div>
  )
}
