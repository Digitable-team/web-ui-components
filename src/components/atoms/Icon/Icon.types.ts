export enum Icons {
  // social
  telegram = 'telegram',
  facebook = 'facebook',
  vkontakte = 'vkontakte',
  twitter = 'twitter',
  github = 'github',

  // common
  gamepad = 'gamepad',
  work = 'work',
  house = 'house',
  star = 'star',
  domain = 'domain',
  graphics = 'graphics',

  // details
  processor = 'processor',
  motherboard = 'motherboard',
  videocard = 'videocard',
  ram = 'ram',
  close = 'close',

  // navLinks
  call = 'call',
  catalog = 'catalog',
  delivery = 'delivery',
  home = 'home',

  // benefit
  dollar = 'dollar',
  guality = 'guality',
  tool = 'tool',
  modules = 'modules',
  vector = 'vector',
  envelope = 'envelope'
}
