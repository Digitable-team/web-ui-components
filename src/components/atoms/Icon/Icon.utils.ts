import { ReactComponent as ArrowAccordion } from '../../../../public/images/ArrowAccordion.svg'
import { ReactComponent as Dollar } from '../../../../public/images/Benefits/Dollar.svg'
import { ReactComponent as Envelope } from '../../../../public/images/Benefits/Envelope.svg'
import { ReactComponent as Guality } from '../../../../public/images/Benefits/Guality.svg'
import { ReactComponent as Modules } from '../../../../public/images/Benefits/Modules.svg'
import { ReactComponent as Tool } from '../../../../public/images/Benefits/Tool.svg'
import { ReactComponent as Vector } from '../../../../public/images/Benefits/Vector.svg'
import { ReactComponent as Github } from '../../../../public/images/Github.svg'
import { ReactComponent as Intro } from '../../../../public/images/Intro.svg'
import { ReactComponent as Logo } from '../../../../public/images/Logo.svg'
import { ReactComponent as LogoAndText } from '../../../../public/images/LogoAndText.svg'
import { ReactComponent as Regru } from '../../../../public/images/Regru.svg'
import { ReactComponent as Telegram } from '../../../../public/images/Telegram.svg'
import { ReactComponent as User } from '../../../../public/images/User.svg'
import { ReactComponent as Vkontakte } from '../../../../public/images/Vk.svg'

import { Icons } from './Icon.types'

export const ICONS: Record<Icons | string, React.FunctionComponent> = {
  [Icons.intro]: Intro,
  [Icons.logo]: Logo,
  [Icons.logoAndText]: LogoAndText,
  [Icons.user]: User,
  [Icons.regru]: Regru,
  [Icons.vkontakte]: Vkontakte,
  [Icons.github]: Github,
  [Icons.telegram]: Telegram,
  [Icons.dollar]: Dollar,
  [Icons.envelope]: Envelope,
  [Icons.guality]: Guality,
  [Icons.modules]: Modules,
  [Icons.tool]: Tool,
  [Icons.vector]: Vector,
  [Icons.arrowAccordion]: ArrowAccordion
}

export const getIcon = (name: Icons | string) => {
  return ICONS[name]
}
