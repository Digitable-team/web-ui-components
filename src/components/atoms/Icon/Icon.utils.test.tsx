import { ICONS } from './Icon.utils'

describe('Icon.utils', () => {
  it('should match snapshot', () => {
    expect(ICONS).toMatchSnapshot()
  })
})
