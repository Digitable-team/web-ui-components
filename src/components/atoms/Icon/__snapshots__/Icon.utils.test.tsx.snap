// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.utils should match snapshot 1`] = `
Object {
  "call": "svg",
  "catalog": "svg",
  "close": "svg",
  "delivery": "svg",
  "dollar": "svg",
  "domain": "svg",
  "envelope": "svg",
  "facebook": "svg",
  "gamepad": "svg",
  "github": "svg",
  "graphics": "svg",
  "guality": "svg",
  "home": "svg",
  "house": "svg",
  "modules": "svg",
  "motherboard": "svg",
  "processor": "svg",
  "ram": "svg",
  "star": "svg",
  "telegram": "svg",
  "tool": "svg",
  "twitter": "svg",
  "vector": "svg",
  "videocard": "svg",
  "vkontakte": "svg",
  "work": "svg",
}
`;
